lib/pbench/test/unit/server/query_apis/test_controllers_list.py

```diff
@@ -12,64 +13,15 @@ class TestControllersList:
     constructor and `post` service.
     """
 
-    def build_index(self, server_config, dates):
-        """
-        Build the index list for query
-
-        Args:
-            dates (iterable): list of date strings
-        """
-        idx = server_config.get("Indexing", "index_prefix") + ".v6.run-data."
-        index = "/"
-        for d in dates:
-            index += f"{idx}{d},"
-        return index
-
-    def test_non_accessible_user_data(self, client, server_config, pbench_token):
-        """
-        Test behavior when Authorization header does not have access to other user's data
-        """
-        # The pbench_token fixture logs in as user "drb"
-        # Trying to access the data belong to the user "pp"
-        response = client.post(
-            f"{server_config.rest_uri}/controllers/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json={"user": "pp", "start": "2020-08", "end": "2020-10"},
+    @pytest.fixture(autouse=True)
+    def _setup(self, client):
+        super()._setup(
+            cls_obj=ControllersList(client.config, client.logger),
+            pbench_endpoint="/controllers/list",
+            elastic_endpoint="/_search?ignore_unavailable=true",
+            payload={"user": "drb", "start": "2020-08", "end": "2020-10"},
+            empty_es_response_payload=Commons.empty_es_response_payload,
```

Comment:
Instead of having the subclasses know the superclass name and specify it here, you could use this as the default of a superclass keyword argument which the subclasses could simply omit.

Reply:
Or, wouldn't `self.empty_es_response_payload` work?

Reply:
Hmmm, I believe that it _would_ work, but it will look extremely funny.

Regardless of how you address it, I recommend that you don't name the instance attribute the same as the class attribute which contains the default value.  While that is sort of clever, I suspect that it will just end up being confusing.  That is, as long as the instance attribute is undefined or unset, any reference to the name will resolve to the class attribute; once the instance attribute has been set, references will resolve to that.  Thus,
```
            empty_es_response_payload=self.empty_es_response_payload,
```
would do exactly what we want it do (copy the class attribute value into the instance attribute), but it will look like a pointless no-op (and, in fact, if we could avoid the assignment altogether, _that_ would do exactly what we want (in this case) it to do, as well...).

So, it would be better if the class attribute were named something like `default_empty_es_response_payload` or `EMPTY_ES_RESPONSE_PAYLOAD` (since it's a "constant").

Reply:
In principle we could just supply the JSON payload as a literal directly to the superclass `_setup` kwarg, just like `payload=` on the line above. Granted, the Elasticsearch verbiage is generally a bit more flowery, and defining it as a subclass attribute would be fine where that gets ugly... which is definitely an uppercased constant. 👍🏻

Reply:
Sounds like consensus.

But, I want to clarify one thing:

> defining it as a subclass attribute would be fine

Since the value is the same across several subclasses, we're defining it as a _super_-class attribute.  (Is that what you meant?)

Reply:
Two separate values, if we're following my earlier suggestion of a superclass "constructor" kwarg:

1. The superclass default, when the kwarg is omitted;
2. The individual (non-default) subclass values which are specified in their superclass setup calls

Either may be an inline literal or, if that's too big, a constant in the local class (or even a local within the setup method since everyone else accesses it only through `self.empty_es_response_payload`).

Reply:
Ah, I see, now.  Yes, I was wondering how we might do something like that -- allowing instances to inherit the class attribute value as their instance attribute value while preserving the ability of subclasses to set it to something else -- and it didn't occur to me to use keyword parameters...so that's...um...quite clever.  But, how would you code that in `_setup()`?  Would you have it do key lookups (e.g., `empty_es_response_payload`) in `kwargs`, and then assign the instance attribute only if the key is present?